### ISSUE TITLE
better inwasm repo integration

### ISSUE THREAD

**Issue by jerch** [xtermjs/xterm.js#4646]
Issues:
- Not sure yet when exactly this happens - sometimes after restarting the watcher on the demo project the *.wasm.js files get rebuilt by the Typescript compiler, leaving the demo project in the error state `error, must run "inwasm" first`. This can be fixed currently by manually navigating to the image addon and calling `yarn inwasm` there. Needs an automated hook in some of the startup script...
- Once we have more wasm logic in the project (see faster serialize addon), the SDK save paths need adjustments to not pull them again for every inwasm invocation. Maybe place the SDK data globally into the base repo...

**Comment by Tyriar** [comment]
The error is pretty annoying 😅, I actually tried to try/catch the import but it ended up resulting in a bunch of changes as the ctor was then only available as a promise.

**Comment by jerch** [comment]
Hmm I can trigger the recompilation from a branch change sometimes while `yarn watch` is still running. Seems the typescript watcher sometimes decides to refresh a bigger portion of the `*.ts` files than actually needed. I dont think we can reliable hook into that, thus a fix script entry for `*.wasm.js` is needed.

**Comment by tisilent** [comment]
![aq1](https://github.com/xtermjs/xterm.js/assets/48614781/c1b4bcf2-ce55-48d2-a7a5-109fa7a4dab6)
😨

**Comment by jerch** [comment]
Yeah sorry for this, the fix is a bit more involved, trying to get it fixed over the weekend.

**Comment by tisilent** [comment]
Not affected.
yarn add inwasm
yarn inwasm
✌️

**Comment by jerch** [comment]
@tisilent 
Ah this worked "by accident" for you - `yarn add some-package` will re-run the init scripts, which also calls `yarn postsetup`.

Also simply calling `yarn inwasm` is not enough - you must at least give it a file matching pattern. To grab all potential wasm files in addon folders this would be `yarn inwasm "addons/*/out/*.wasm.js"`. But dont do that yet - it will pull SDKs and stuff for no good reason cluttering your repo folder. I need to fix https://github.com/jerch/inwasm/issues/20 first, then I can add a more appropriate script to package.json to get `inwasm` out of the way.

Until then calling `yarn postsetup` is your best bet, as @Tyriar pointed out.

**Comment by Tyriar** [comment]
I got pretty confused when trying to run `yarn test-api`, it gives this error:

```
1) WebLinksAddon
       "before all" hook for ".com":
     page.evaluate: ReferenceError: Terminal is not defined
    at eval (eval at evaluate (:201:30), <anonymous>:1:19)
    at eval (<anonymous>)
    at UtilityScript.evaluate (<anonymous>:201:30)
    at UtilityScript.<anonymous> (<anonymous>:1:44)
      at C:\Github\Tyriar\xterm.js\test\api\TestUtils.ts:47:14
      at step (out-test\api\TestUtils.js:44:23)
      at Object.next (out-test\api\TestUtils.js:25:53)
      at C:\Github\Tyriar\xterm.js\out-test\api\TestUtils.js:19:71
      at __awaiter (out-test\api\TestUtils.js:15:12)
      at openTerminal (out-test\api\TestUtils.js:123:12)
      at Context.<anonymous> (addons\xterm-addon-web-links\test\WebLinksAddon.api.ts:39:23)
      at fulfilled (addons\xterm-addon-web-links\out-test\WebLinksAddon.api.js:5:58)
```

**Comment by jerch** [comment]
@Tyriar Sure thats caused by `inwasm`? The `page.evaluate: ReferenceError: Terminal is not defined` looks strange to me, is there some exception handling swallowing the real exception behind it?

**Comment by Tyriar** [comment]
Running inwasm fixed it, the exception is thrown in the demo page imports so it doesn't get to the `window.Terminal` assignment

**Comment by jerch** [comment]
Well my planned fix for inwasm just creates the ability to add a call to watcher scripts in the end - I am not sure if that will be enough to mitigate all the issues seen so far.

If this still fails I fear we have to move the inwasm parts out of the base repo, and import things from our own third-party package only containing those wasm definitions. :cry:

**Comment by jerch** [comment]
As an interim fix, lets move `inwasm` out of the way (#4720).
